static/js/containers/App.js

```diff
@@ -1,24 +1,51 @@
 // @flow
 import React from "react"
 import { Route } from "react-router-dom"
+import { connect } from "react-redux"
 
 import HomePage from "./HomePage"
 import ChannelPage from "./ChannelPage"
 import PostPage from "./PostPage"
 import AdminPage from "./admin/AdminPage"
 import CreatePostPage from "./CreatePostPage"
+import Toolbar from "../components/Toolbar"
+import Drawer from "../containers/Drawer"
+
+import { actions } from "../actions"
+import { setShowDrawer } from "../actions/ui"
+import { setChannelData } from "../actions/channel"
 
 import type { Match } from "react-router"
+import type { Dispatch } from "redux"
 
-export default class App extends React.Component {
+class App extends React.Component {
   props: {
-    match: Match
+    match: Match,
+    showDrawer: boolean,
+    dispatch: Dispatch
+  }
+
+  toggleShowSidebar = () => {
+    const { dispatch, showDrawer } = this.props
+    dispatch(setShowDrawer(!showDrawer))
+  }
+
+  componentWillMount() {
+    this.loadData()
+  }
+
+  loadData = async () => {
+    const { dispatch } = this.props
+    const channels = await dispatch(actions.subscribedChannels.get())
+    dispatch(setChannelData(channels))
   }
 
   render() {
     const { match } = this.props
     return (
       <div className="app">
+        <Toolbar toggleShowSidebar={this.toggleShowSidebar} />
+        <Drawer />
```

Comment:
No, I think we want to always render the `<Toolbar />`, I'm not sure why we would want to hide it.

Reply:
also the `<Drawer />` I think we need to basically always render, even if it is hidden, because the material components manage their own state (ask me how I feel about *that* design decision...) so we have to render it even if it is closed.

Reply:
Ah, misread the css rule for `.navbar`. I thought we were hiding the whole thing for desktop widths

Reply:
yeah, we're just hiding the 'menu' button on the toolbar, so you can't open the drawer on a desktop width.